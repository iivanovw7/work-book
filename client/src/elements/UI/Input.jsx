import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import { inputsBackground } from '../../theme';
/* eslint react/require-default-props: 0 */

const StyledInput = styled.input`
  ${inputsBackground};
  padding: 0.5em;
  margin-top: 0.5em;
  border: none;
  border-radius: 3px;
  width: 100%;
`;

const StyledSpan = styled.span`
  color: red;
`;

const StyledLabel = styled.label`
  width: 100%;
  margin-top: 2em;
`;

const Input = (props) => {
  const inputRef = React.createRef();
  const {
    label, validate, id, type
  } = props;

  return (
    <StyledLabel
      ref={inputRef}
      htmlFor={id}
      onMouseEnter={() => {
        inputRef.current.focus();
      }}
    >
      {label}
      {'  '}
      <StyledSpan>{validate}</StyledSpan>
      <StyledInput theme={{ mode: 'dark' }} type={type} id={id} {...props} />
    </StyledLabel>
  );
};

Input.propTypes = {
  label: PropTypes.string.isRequired,
  validate: PropTypes.string,
  id: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired
};

export default Input;
