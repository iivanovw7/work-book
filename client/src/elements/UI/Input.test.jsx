import 'cross-fetch/polyfill';
import 'jsdom-global/register';
import React from 'react';
import { mount } from 'enzyme';
import { Input } from './Input';

jest.mock('../../config/apiURL', () => () => '/api');

describe('Testing Input component', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  const props = {
    label: 'label',
    validate: 'validate',
    type: 'type',
    id: '1'
  };

  const Composition = () => (
    <Input userAccess {...props} />
  );

  it('Should match snapshot', () => {
    const component = mount(<Composition />);

    expect(component).toMatchSnapshot();
  });

  it('Should render with all child elements', () => {
    const component = mount(<Composition />);

    expect(component.find('span')).toHaveLength(1);
    expect(component.find('input')).toHaveLength(1);
    expect(component.find('label')).toHaveLength(1);

    expect(component.find('span').text()).toBe('validate');
    expect(component.find('label').text()).toBe('label  validate');
  });
});
