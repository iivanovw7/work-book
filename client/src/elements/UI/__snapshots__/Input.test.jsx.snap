// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input component Should match snapshot 1`] = `ReactWrapper {}`;
